docs/reference.rst

```diff
@@ -103,6 +127,21 @@ Helper objects
     :members:
     :member-order: bysource
 
+.. autoclass:: pygfx.PointLightHelper
+    :members:
+    :member-order: bysource
+
+.. autoclass:: pygfx.DirectionalLightHelper
+    :members:
+    :member-order: bysource
+
+.. autoclass:: pygfx.SpotLightHelper
+    :members:
+    :member-order: bysource
+
+.. autoclass:: pygfx.DirectionalLightShadowHelper
```

Comment:
Er, this class is provided because directional light shadows are a little special. This is also one of the reasons why `position` and `target` are used instead of `direction` in the `DirectionalLight`'s API.

Ideally, directional light affects the whole scene, has only direction and no position. However, we need a camera (from the perspective of light) to render shadow maps, so it must have a position, and only the part of scene within the shadow camera's view frustum can generate shadow maps. The resolution of shadow map is certain, and the view frustum of shadow camera is also limited. If the view frustum of shadow camera is too large, the shadow sampling accuracy of nearby objects in scene will be too low.

This tool class is designed to help users set reasonable parameters to generate appropriate directional light shadows. The specific effect can be seen in the `light_directional_shadow.py` example.

If we want to realize the position independent global directional light shadow, we need to automatically calculate the appropriate viewing cone of the shadow camera (for the orthogonal camera, the direction has been determined, that is, we need to calculate the appropriate camera position), so that it can exactly include the current display scene, and use the CSM(Cascaded Shadow Mapping) algorithm to generate shadow maps in different regions. Maybe we can add these advanced functions later.

For shadow generation, directional light is more troublesome, while spotlight is relatively simple. The shadow camera parameters can be completely determined by the parameters of spotlight itself.

Sorry for my less proficient English, I hope I can explain it clearly.  :sweat_smile: